packages/app/src/cli/models/app/app.ts

```diff
@@ -67,6 +67,8 @@ export const AppConfigurationSchema = zod.union([LegacyAppSchema, AppSchema])
  */
 export type AppConfiguration = zod.infer<typeof AppConfigurationSchema> & {path: string}
 
+export type AppConfigurationWithoutPath = zod.infer<typeof AppConfigurationSchema>
```

Comment:
It is a bit tricky. AppConfiguration encompasses having an app that is in the legacy/template state (the schema at the start of that file) as well as one in the more familiar form. Similarly an app/config can be loaded into that state too -- `AppInterface` wraps up config that can be in either format.

I'm trying to chip away at this and change so that the `Current...` config is the one most code sees, as in reality its a very narrow circumstance where we would actually want to work with with one of those old-style files now. The work in the `REDACTED` project will include (I think) a more opinionated app assembly process, where the constructed app is only ever in the up to date format. I'd like to see what breaks with that -- hopefully not much. Then we can simplify a hell of a lot of this.

As-is though, this is the right type to use because the config _can_ be in either format today.